# STAT 207 Group Lab Assignment 11 - [10 total points]

## Logistic Regression

<hr>

## <u>Lab Grading</u>:

Should we grade your submission?  If not, write the netID of the submission to be graded.  (Note: We will only grade one assignment per group, and we'll pick the first one that says we should grade that submission.  We will assign the same grade to all team members.)

*For example*, you might respond: **grade this submission** or **my submission is under netID jdeeke**

mdm16

If you said **my submission is under netID** above, we will not read any more of your lab submission.

If you said **grade this submission** above, who worked with you on this submission?  Write both their **names** and **netIDs**.  Also, discuss and record if you follow the NBA + your favorite NBA team, if you have one.

<hr>

## <u>Purpose</u>:
You should work in groups of 2-3 on this report (not working in groups without permission will result in a point deduction). The purpose of this group lab assignment is to fit and use a logistic regression model.
<hr>

## <u>Assignment Instructions</u>:

### Group Roles

Suggested and specified roles are provided below: 

#### Groups of 2

* **Driver**: This student will type the report.  While typing the report, you may be the one who is selecting the functions to apply to the data.
* **Navigator**: This student will guide the process of answering the question.  Specific ways to help may include: outlining the general steps needed to solve a question (providing the overview), locating examples within the course notes, and reviewing each line of code as it is typed.

#### Groups of 3

* **Driver**: This student will type the report.  They may also be the one to select the functions to apply to the data.
* **Navigator**: This student will guide the process of answering the question.  They may select the general approach to answering the question and/or a few steps to be completed along the way. 
* **Communicator**: This student will review the report (as it is typed) to ensure that it is clear and concise.  This student may also locate relevant examples within the course notes that may help complete the assignment.

<hr>

### Imports

In [1]:
#Run this
import pandas as pd                    # imports pandas and calls the imported version 'pd'
import matplotlib.pyplot as plt        # imports the package and calls it 'plt'
import seaborn as sns                  # imports the seaborn package with the imported name 'sns'
sns.set()  

## Case Study: Understanding NBA Longevity

We will look at data collected on rookies in the NBA from 1980-2016 to determine if the rookies are successful and still playing in the NBA after 5 years.  The available variables include the following **per game** statistics:

- **GP**: number of games played
- **MIN**: number of minutes played per game
- **PTS**: number of points scored per game
- **FGM**: number of field goals made per game
- **FGA**: number of field goals attempted per game
- **FG%**: percentage of field goals made per game
- **3P Made**: number of 3-point field goals made per game
- **3PA**: number of 3-point field goals attempted per game
- **3P%**: percentage of 3-point field goals made per game
- **FTM**: number of free throws made per game
- **FTA**: number of free throws attempted per game
- **FT%**: percentage of free throws made per game
- **OREB**: number of offensive rebounds per game
- **DREB**: number of defensive rebounds per game
- **REB**: total number of rebounds per game
- **AST**: number of assists per game
- **STL**: number of steals per game
- **BLK**: number of blocks per game
- **TOV**: number of turnovers per game
- **TARGET_5Yrs**: whether the player is still playing in the NBA after 5 years (1 = playing, 0 = not playing)

The code cell below will read in the data for you.  Be sure to run the cell. 

In [8]:
df = pd.read_csv('nba_rookie.csv')
df.columns

Index(['Name', 'GP', 'MIN', 'PTS', 'FGM', 'FGA', 'FG_Perc', '3P Made', '3PA',
       '3P_Perc', 'FTM', 'FTA', 'FT_Perc', 'OREB', 'DREB', 'REB', 'AST', 'STL',
       'BLK', 'TOV', 'TARGET_5Yrs'],
      dtype='object')

### 1. [5.5 points] Fit a Logistic Regression Model

**a)** Fit a model predicting whether the player lasts 5 years or more in the NBA from four of the other variables in the data.  You may pick the combination of four variables to use as predictors.

In [5]:
import statsmodels.api as sm
X = df[['PTS','TOV', 'MIN', 'AST']]
X = sm.add_constant(X) #adds a constant for intercept
y = df['TARGET_5Yrs']
log_model = sm.Logit(y,X).fit()
print(log_model.summary())

Optimization terminated successfully.
         Current function value: 0.591365
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:            TARGET_5Yrs   No. Observations:                 1309
Model:                          Logit   Df Residuals:                     1304
Method:                           MLE   Df Model:                            4
Date:                Fri, 10 Apr 2026   Pseudo R-squ.:                  0.1029
Time:                        11:18:58   Log-Likelihood:                -774.10
converged:                       True   LL-Null:                       -862.93
Covariance Type:            nonrobust   LLR p-value:                 2.368e-37
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const         -1.0148      0.162     -6.251      0.000      -1.333      -0.697
PTS            0.1568      0.

**b)** What is the pseudo-$R^2$ for this model?  

In [6]:
print(log_model.prsquared)

0.102942043363904


**c)** In general, what is the pseudo-$R^2$ measuring?  In other words, what quality of a model is measured with this number?

Pseudo-R^2 measures how much better the logistic regression model fits the data compared to a model with no predictors. Larger values mean the predictors do a better job explaining/prediciting whether a rookie lasts five years.

**d)** Interpret the coefficient fit for one of the predictor variables in terms of the **odds**.

The coefficient for PTS is .1568. This corresponds to an e^(.1568) or 1.17 change in odds for each addition 1 point per game. ie, For each addition 1 point per game the odds that a player lasts 5 years in the NBA is multiplied by about 1.17

**e)** What are the baseline odds for the fitted model?

baseline odds is just the e raised to intercept (e^(-1.1048)) = .36. So baseline odds for the fitted model is .36

### 2. [2 points] Ayo Donsunmu

A recent player in the NBA from the University of Illinois is Ayo Donsunmu.  

**a)** Use the NBA Regular Season Stats per game for the 2021-2022 season (Ayo's rookie season) from this website https://basketball.realgm.com/player/Ayo-Dosunmu/Summary/107970 to predict the probability that Ayo will last in the NBA for at least 5 years.  Be sure to scroll down and use his statistics reported for the 2021-2022 season, corresponding to his rookie year.

In [9]:
#Stats:    MIN = 27.4, PTS = 8.8 AST = 3.3 TOV = 1.4
new_player = pd.DataFrame({
    'const': [1],
    'PTS': [8.8],
    'TOV': [1.4],
    'MIN': [27.4],
    'AST': [3.3]
})

log_model.predict(new_player)

0    0.764539
dtype: float64

**b)** Based on the available observations in the dataset, does this prediction seem trustworthy?  Briefly explain.

*Hint*: there are two ways that you can answer this question, both of which compare Ayo's characteristics to those in the available data.  One approach requires analysis of the data, while the other does not.

### 3. [2.5 points] Appropriate Model?

**a)** Is a logistic regression model the appropriate model for the data?  Or, should we use linear regression instead?  Briefly explain.

**b)** Generate and interpret a plot to assess the linearity assumption for this model.  Does this model seem to be an appropriate model for the data?

### 4. AI Acknowledgement

Our course policy is that you should write all of your own interpretations and other narrative answers (phrases or sentences) yourself without the assistance of AI.  AI should not be used on Questions 1c, 1d, 2b, 3a, and the explanation for 3b of this assignment.  

You may use AI to help guide your code, although you should write all of your own code yourself (not copy-paste from another source) and you should cite your use of AI.  I would encourage you to try to generate any necessary code yourself first using course resources and using AI as a debugging tool if/when you reach an error that you can't figure out or to help you perform any coding tasks that are more advanced than we've demonstrated during class (intended only for projects).  

Did you use AI on this assignment?  Did you use other resources outside of our course-provided resources on this assignment?

If you used AI or other resources, answer the following questions to cite your usage.

- Which AI and/or resources did you use?
- What prompts did you ask it?
- How did you integrate the responses into your assignment?  Specifically, which questions or parts are associated with this usage?

Note: answering these three questions are enough for our course but may not be enough for a different course or context.